server/lib/subscriptions.js

```diff
@@ -253,6 +265,24 @@ export async function cancelSubscriptionAndNotifyUser(order) {
   return sendFailedEmail(order, true);
 }
 
+/** Send `archived.collective` email */
+export async function notifyUserForArchivedCollective(order) {
+  const user = order.createdByUser;
```

Comment:
Unfortunately we don't have any database constraint to ensure that `createdByUser` is not null. Though we only have one order like that in the database, we need to be defensive here to ensure we don't crash subscriptions if we end up with one.

Reply:
Agree, let's rely on information provided by the `order.fromCollective` (or similar, not sure about the syntax).

Reply:
`order.createdByUser`  is used by other methods within that file too, basically to access `generateLoginLink` method and user `email` but I don't think `order.fromCollective` has the method nor the `email`. But one way to we can do it is to query `user` with `order.fromCollective.id`, what do you think?

Reply:
You're right, `order.createdByUser` is the current practice and there is no point in refactoring that here. We should however keep that in mind, and later refactor to notify **accounts: (collectives/organizations/individuals)** and not **users**.